pkg/model/stack_expander.go

```diff
@@ -8,6 +8,42 @@ import (
 	yaml3 "gopkg.in/yaml.v3"
 )
 
+func isMapString(value string) bool {
+	colonSplit := strings.SplitN(value, ":", 3)
+	switch len(colonSplit) {
+	case 2:
+		indxColon := strings.Index(value, ":")
+		indxRightCurlyBracket := strings.Index(value, "}")
+		return indxColon > indxRightCurlyBracket
+	case 3:
+		return true
+	}
+	return false
+}
+func isCurlyEnv(value string) bool {
+	return strings.HasPrefix(value, "{") && strings.HasSuffix(value, "}")
+}
+func hasEnvDefaultValue(value string) (int, bool) {
+	return strings.Index(value, ":-"), strings.Contains(value, ":-")
+}
```

Comment:
Can we create simple unit tests for these functions? I see that the env expansion starts to relies in this and including tests will help us to prevent issues in the future if we start to detect o include new cases